lib/decompress/zstd_decompress_block.c

```diff
@@ -1073,9 +1161,11 @@ MEM_STATIC void ZSTD_assertValidSequence(
 #endif
 
 #ifndef ZSTD_FORCE_DECOMPRESS_SEQUENCES_LONG
+
+
 FORCE_INLINE_TEMPLATE size_t
```

Comment:
my "gut feeling" is that this variant should not be inlined.
It's a detrimental corner case, that should be avoided as much as possible. And when it cannot, well it's clear that it's going to be a bit slower, and that's a cost which is accepted.
If this function is inlined, its impact may be felt beyond its own immediate scope, and trickle down through compiler optimization into other parts of the code, which are considered more performance sensitive.